dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/handler/DataHandler.java

```diff
@@ -985,6 +994,24 @@ private void handleEmptyDimensionItemPermutations( List<List<DimensionItem>> dim
         return result;
     }
 
+    private Grid getOrFetchGrid( final DataQueryParams params, final Function<DataQueryParams, Grid> function )
```

Comment:
I think this method is already in `AnalyticsCache`#`getOrFetch` ? So we don't need to replicate it here?

Reply:
ups, just tested something and forget to remove, sorry. done

Reply:
sorry, just tested something and forget to remove, it is done.